tests/testthat/test-term-directed.R

```diff
@@ -0,0 +1,299 @@
+  expect_equal(s.att, c(439,2345), ignore_attr=TRUE)
+})
+
+test_that("nodeifactor, directed", {
+  s.a <- summary(samplike~nodeifactor("group"))
+  e.a <- ergm(samplike~nodeifactor(~group), estimate="MPLE")
+  s.ab <- summary(samplike~nodeifactor(function(x) x %v% "Trinity", levels=TRUE))
+  e.ab <- ergm(samplike~nodeifactor("Trinity", base=(2:3)), estimate="MPLE")
+  expect_summary(s.a, e.a, c(13, 46), -c(1.4424, .4618))
+  expect_summary(s.ab, e.ab, c(28, 29, 31), -.9719)
+})
+
+test_that("nodeocov, directed", {
+  s.a <- summary(samplike~nodeocov("YearsServed"))
+  e.a <- ergm(samplike~nodeocov(function(x) x %v% "YearsServed"), estimate="MPLE")
+  s.at <- summary(samplike~nodeocov(~YearsServed^2))
+  e.at <- ergm(samplike~nodeocov(~(.%v%"YearsServed")^2), estimate="MPLE")
+  s.att <- summary(samplike~nodeocov(~poly(YearsServed,2,raw=TRUE)))
+  expect_summary(s.a, e.a, 467, -.1581)
+  expect_summary(s.at, e.at, 2691, -.02243)
+  expect_equal(s.att, c(467, 2691), ignore_attr=TRUE)
+})
+                
+test_that("nodeofactor, directed", {
+  s.a <- summary(samplike~nodeofactor("group"))
+  e.a <- ergm(samplike~nodeofactor(~group), estimate="MPLE")
+  s.ab <- summary(samplike~nodeofactor("Trinity", levels=TRUE))
+  e.ab <- ergm(samplike~nodeofactor(function(x) x %v% "Trinity", base=(2:3)), estimate="MPLE")
+  expect_summary(s.a, e.a, c(18,36), -c(1.0217, .8353))
+  expect_summary(s.ab, e.ab, c(31,30,27), -.8287)
+})
+
+test_that("odegree, directed", {
+  s.d <- summary(samplike~odegree(2:3))
+  e.d <- ergm(samplike~odegree(3), estimate="MPLE")
+  s.db <- summary(samplike~odegree(1:3, function(x) x %v% "group"))
+  e.db <- ergm(samplike~odegree(4, "group"), estimate="MPLE")
+  s.dbh <- summary(samplike~odegree(4:5, ~group, TRUE))
+  e.dbh <- ergm(samplike~odegree(2, "group", TRUE), estimate="MPLE")
+  expect_summary(s.d, e.d, c(0,1), -.1625189)
+  expect_summary(s.db, e.db, c(0,0,0,0,0,1,0,0,0), -c(-1.6292, 0.1112, 0.1625))
+  expect_summary(s.dbh, e.dbh, c(6,3), -1.344)
+})
+
+test_that("ostar, directed", {
+  s.k <- summary(samplike~ostar(1:3))
+  e.k <- ergm(samplike~ostar(c(2,4)), estimate="MPLE")
+  s.ka <- summary(samplike~ostar(2, "group"))
+  e.ka <- ergm(samplike~ostar(2, function(x) x %v% "group"), estimate="MPLE")
+  expect_summary(s.k, e.k, c(88,178, 191), c(.1224, -.1986))
+  expect_summary(s.ka, e.ka, 88, .1466)
+})
+                     
+test_that("odegree1.5, directed", {
+  s.0 <- summary(samplike~odegree1.5)
+  e.0 <- ergm(samplike~odegree1.5, estimate="MPLE")
+  expect_summary(s.0, e.0, 196.9432, -0.2909)
+})
+
+test_that("odegrange, directed", {
+  s.0 <- summary(samplike~odegrange(5:8))
+  e.0 <- ergm(samplike~odegrange(5:8), estimate="MPLE")
+  s.h <- summary(samplike~odegrange(5:8, by=function(x) x %v% "group", homophily=TRUE))
+  e.h <- ergm(samplike~odegrange(5:8, by="group", homophily=TRUE), estimate="MPLE")
+  expect_summary(s.0, e.0, c(12, 5, 0, 0), -c(0.619, -1.030, Inf, Inf))
```

Comment:
The check on the coefficients fail with

> Failure (test-term-directed.R:235:3): odegrange, directed                      
> coef(e) (`actual`) not equal to `coefficients` (`expected`).                   
>                                                                                
>   `actual`: -0.61904 -1.02962 -Inf -Inf                                        
> `expected`: -0.61900  1.03000 -Inf -Inf                                                                                                                        
> Backtrace:                             
>  1. ergm:::expect_summary(...) test-term-directed.R:235:2                      
>  2. testthat::expect_equal(...) test-term-directed.R:14:2                      

The relevant check in the old code is:
`if (... || round(coef(e.0) + c(0.619, -1.030, Inf, Inf))!= 0 ||...) {
	print(list(s.0=s.0, e.0=e.0, s.h=s.h, e.h=e.h))
	stop("Failed odegrange term test")`

Reply:
Does this work if ran standalone but fail if run inside `testthat`?

Reply:
I commented out all the other test cases and sourced that file and the checking logic and sourced the file. Confirmed that the output is `c(-0.619, -1.03, -Inf, -Inf)`. What I think is happening is that this (and some other checks in the file) is missing an `all`. Does it need a closer look? I can update the reference value otherwise.

Reply:
Indeed, as implemented, it would only check the match for the first element. Good catch!

Please do check for similar issues in the other code.